kimchi/src/prover.rs

```diff
@@ -305,32 +320,22 @@ where
         let gamma = fq_sponge.challenge();
 
         //~ 13. If using lookup:
-        if let Some(lcs) = &index.cs.lookup_constraint_system {
+        if index.cs.lookup_constraint_system.is_some() {
             //~     - Compute the lookup aggregation polynomial.
-            let iter_lookup_table = || {
-                (0..d1_size).map(|i| {
-                    let row = lcs.lookup_table8.iter().map(|e| &e.evals[8 * i]);
-                    let table_id = match lcs.table_ids8.as_ref() {
-                        Some(table_ids8) => table_ids8.evals[8 * i],
-                        None =>
-                        // If there is no `table_ids8` in the constraint system, every
-                        // table ID is identically 0.
-                        {
-                            ScalarField::<G>::zero()
-                        }
-                    };
-                    combine_table_entry(
-                        &lookup_context.joint_combiner.unwrap(),
-                        &lookup_context.table_id_combiner.unwrap(),
-                        row,
-                        &table_id,
-                    )
-                })
+            let joint_lookup_table_d8 = lookup_context.joint_lookup_table_d8.as_ref().unwrap();
+
+            let joint_lookup_table = {
```

Comment:
ignore this block, as I only did this to see if the tests would still pass. If I get a 👍 for the approach taken in this PR I'll implement the changes to `aggregation()` in order to leverage this `joint_lookup_table_d8` (the same way I modified `sorted()`)